tests/common/dualtor/dual_tor_utils.py

```diff
@@ -87,4 +93,244 @@ def get_t1_ptf_ports(dut, tbinfo):
+
+    Args:
+        tor_host (object): Host object for the ToR DUT.
+        tor_fanouthosts (dict): Key is fanout hostname, value is fanout host object.
+        tbinfo (dict): Testbed info from the tbinfo fixture.
+        dut_intfs (list, optional): List of DUT interface names, for example: ['Ethernet0', 'Ethernet4']. All the
+            fanout interfaces that are connected to the specified DUT interfaces will be shutdown. If dut_intfs is not
+            specified, the function will shutdown all the fanout interfaces that are connected to the tor_host DUT.
+            Defaults to None.
+
+    Returns:
+        list of tuple: Return a list of tuple. Each tuple has two items. The first item is the host object for fanout.
+            The second item is the fanout interface that has been shutdown. The returned list makes it easy to recover
+            the interfaces.
+    """
+    down_intfs = []
+
+    if not dut_intfs:
+        # If no interface is specified, shutdown all ports
+        mg_facts = tor_host.get_extended_minigraph_facts(tbinfo)
+        dut_intfs = mg_facts['minigraph_ports'].keys()
+
+    dut_intfs = natsorted(dut_intfs)
+
+    full_dut_fanout_port_map = {}
+    for _, fanout_host in tor_fanouthosts.items():
+        for encoded_dut_intf, fanout_intf in fanout_host.host_to_fanout_port_map.items():
+            full_dut_fanout_port_map[encoded_dut_intf] = {
+                'fanout_host': fanout_host,
+                'fanout_intf': fanout_intf
+            }
+
+    logger.debug('full_dut_fanout_port_map: {}'.format(full_dut_fanout_port_map))
+
+    for dut_intf in dut_intfs:
+        encoded_dut_intf = encode_dut_port_name(tor_host.hostname, dut_intf)
+        if encoded_dut_intf in full_dut_fanout_port_map:
+            fanout_host = full_dut_fanout_port_map[encoded_dut_intf]['fanout_host']
+            fanout_intf = full_dut_fanout_port_map[encoded_dut_intf]['fanout_intf']
+            fanout_host.shutdown(fanout_intf)
+            down_intfs.append((fanout_host, fanout_intf))
+        else:
+            logger.error('No dut intf "{}" in full_dut_fanout_port_map'.format(encoded_dut_intf))
+
+    return down_intfs
+
+
+@pytest.fixture
+def shutdown_fanout_upper_tor_intfs(upper_tor_host, upper_tor_fanouthosts, tbinfo):
+    """Fixture for shutting down fanout interfaces connected to specified upper_tor interfaces.
+
+    Args:
+        upper_tor_host (object): Host object for upper_tor.
+        upper_tor_fanouthosts (dict): Key is fanout hostname, value is fanout host object.
+        tbinfo (dict): Testbed info from the tbinfo fixture.
+
+    Yields:
+        function: A function for shutting down fanout interfaces connected to specified upper_tor interfaces
+    """
+    down_intfs = []
+
+    def shutdown(dut_intfs=None):
+        _down_intfs = shutdown_fanout_tor_intfs(upper_tor_host, upper_tor_fanouthosts, tbinfo, dut_intfs)
+        for fanout_host, fanout_intf in _down_intfs:
```

Comment:
```suggestion
        down_intfs.extend(_down_intfs)
```

Nit: `extend` may be more concise

Reply:
Thanks! Updated.